cmake/std/trilinosprhelpers/setenvironment/test/test_ModuleHelper.py

```diff
@@ -0,0 +1,150 @@
+    def communicate(self):
+        print("mock_popen> communicate()")
+        stdout = b"stdout=1"
+        stderr = b"stderr=2"
+        self.returncode = 0
+        return (stdout,stderr)
+
+
+
+class mock_popen_status_ok(mock_popen):
+    """
+    Specialization of popen mock that will return with success.
+    """
+    def __init__(self, cmd, stdout=None, stderr=None):
+        super(mock_popen_status_ok, self).__init__(cmd,stdout,stderr)
+
+
+
+class mock_popen_status_error(mock_popen):
+    """
+    Specialization of popen mock that will return with success.
+    """
+    def __init__(self, cmd, stdout=None, stderr=None):
+        super(mock_popen_status_error, self).__init__(cmd,stdout,stderr)
+
+    def communicate(self):
+        print("mock_popen> communicate()")
+        stdout = b"stdout=1"
+        stderr = b"ERROR: Something wrong happened."
+        self.returncode = 1
+        return (stdout,stderr)
+
+
+class mock_popen_status_error_mlstatus(mock_popen):
+    """
+    Specialization of popen mock that will return with success.
+    """
+    def __init__(self, cmd, stdout=None, stderr=None):
+        super(mock_popen_status_error_mlstatus, self).__init__(cmd,stdout,stderr)
+
+    def communicate(self):
+        print("mock_popen> communicate()")
+        stdout = b"stdout=1"
+        stderr = b"_mlstatus = False"
+        self.returncode = 1
+        return (stdout,stderr)
+
+
+
+class moduleHelperTest(TestCase):
+    """
+    Main test driver for the module() function provided by the
+    ModuleHelper.py file
+    """
+    def setUp(self):
+        pass
+
+
+    @patch('subprocess.Popen', side_effect=mock_popen_status_ok)
+    def test_module_load_status_ok(self, arg_popen):
+        r = ModuleHelper.module("load", "sems-gcc/4.8.4")
+        print("result = {}".format(r))
+        self.assertEqual(0, r)
+
+
+    @patch('subprocess.Popen', side_effect=mock_popen_status_error)
+    def test_module_load_status_error(self, arg_popen):
+        r = ModuleHelper.module("load", "sems-gcc/4.8.4")
+        print("result = {}".format(r))
+        self.assertEqual(1, r)
+
+
+    @patch('subprocess.Popen', side_effect=mock_popen_status_ok)
+    def test_module_swap_status_ok(self, arg_popen):
+        r = ModuleHelper.module("swap", "sems-gcc/4.8.4", "sems-gcc/7.3.0")
+        print("result = {}".format(r))
+        self.assertEqual(0, r)
+
+
+    #@patch('subprocess.Popen', side_effect=mock_popen_status_ok)
+    def test_module_unload_status_ok(self):
+        with patch('subprocess.Popen', side_effect=mock_popen_status_ok):
+            r = ModuleHelper.module("unload", "sems-gcc/4.8.4")
+        print("result = {}".format(r))
+        self.assertEqual(0, r)
+
+
+    def test_module_load_args_as_list(self):
+        with patch('subprocess.Popen', side_effect=mock_popen_status_ok):
+            r = ModuleHelper.module(["unload", "sems-gcc/4.8.4"])
+        print("result = {}".format(r))
+        self.assertEqual(0, r)
```

Comment:
Not sure the test is doing what the name suggests, but maybe I'm just reading it wrong.

Reply:
```python
module( ["unload", "sems-gcc/4.8.4"] )
#       ^                          ^ - list.
```

Reply:
Oh, gotcha.  The module subcommand (load vs unload) doesn't matter because it's testing passing the args in as a list.  👍